File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.java
class Solution {
    private int[] arr;
    private int tar;
    private int count = 0;

    public int solution(int[] numbers, int target) {
        arr = numbers;
        tar = target;

        dfs(0, 0);

        return count;
    }

    void dfs(int index, int sum) {
        if (index == arr.length) {
            if (sum == tar) {
                count++;
            }
            return;
        }

        dfs(index + 1, sum + arr[index]);
        dfs(index + 1, sum - arr[index]);
    }
}

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py
class Main:
    numbers = []
    target = 0

    def __init__(self, numbers, target):
        self.numbers = numbers
        self.target = target

    def recursive(self, li):
        if len(li) == len(self.numbers):
            if sum(li) == self.target:
                return 1
            else:
                return 0

        return self.recursive([*li, self.numbers[len(li)]]) + self.recursive(
            [*li, self.numbers[len(li)] * -1]
        )


def solution(numbers, target):
    return Main(numbers, target).recursive([])
